src/coreclr/jit/lowerxarch.cpp

```diff
@@ -3414,6 +3414,37 @@ GenTree* Lowering::LowerHWIntrinsicCndSel(GenTreeHWIntrinsic* node)
             blendVariableId = NI_EVEX_BlendVariableMask;
             op1             = maskNode;
         }
+        else if (op2->IsVectorZero() || op3->IsVectorZero())
+        {
+            // If either of the value operands is const zero, we can optimize down to AND or AND_NOT.
+            GenTree* binOp = nullptr;
+
+            if (op3->IsVectorZero())
+            {
+                binOp = comp->gtNewSimdBinOpNode(GT_AND, simdType, op1, op2, simdBaseJitType, simdSize);
+                BlockRange().Remove(op3);
+            }
+            else
+            {
```

Comment:
Is it plausible, meaningful, and worth testing _both_ `op2` and `op3` are both  `IsVectorZero()`?